packages/framework/memo-cache/README.md

```diff
@@ -0,0 +1,141 @@
+
+#### Object Identity
+
+The other benefit to this pattern is maintaining object identity between subsequent calls. In react-native the object identity of the style property will sometimes be compared, even if the values within are identical, the shallow props compare will not see the objects as the same.
+
+Similarly if a style is being turned into a CSS class (which is expensive), a `WeakMap` to map style objects to CSS classes will only work if the object identities are maintained.
+
+## Usage guide
+
+The baseline cache pattern is defined by the following type:
+
+```ts
+export type GetMemoValue<T, TGet = any> = (
+  factory: T | () => T,
+  keys: any[]
+) => [T, GetMemoValue<TGet>];
+```
+
+The parameters are used as follows:
+
+| Param     | Description                                                                                                                                                                                                                                                                             |
+| --------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `factory` | This is typically a function, often a simple closure, which returns a value. This value will be cached for a given set of keys. Subsequent calls will just return the value without executing the function. This can also be a value type in which case that will be returned directly. |
+| `keys`    | Variable parameter list, used as the keys for caching                                                                                                                                                                                                                                   |
+
+The return result is a tuple with two elements containing:
+
+- The result of the factory function, or the value of factory if it is not a function
+- A function for caching which is local to the previous keys queried
+
+This recursive calling pattern can be used to create branches piece by piece in component execution. See the examples below for more detail.
+
+### getMemoCache
+
+To get an instance of the memo cache to work with, a caller starts by calling `getMemoCache`.
+
+```ts
+export function getMemoCache<T = any>(key?: object): GetMemoValue<T>;
+```
+
+This function takes an optional parameter `key` which can be an object to use as a base cache reference.
```

Comment:
Perhaps a more descriptive param name would be `globalCache`. Also if it serves as a base cache reference, is there some type guarantees it needs to make (haven't gotten very far through this review yet)

Reply:
Looks like it doesn't need any type gurantees

Reply:
Hmm...maybe weakKey or globalKey or something like that.  It is simply an identify reference and can be any object (including arrays and functions) and will be weakly held.